### PR TITLE
Always use local yarn cache with yarn 2

### PR DESCRIPTION
**Pull request by brandonchinn178**
Fixes https://github.com/heroku/heroku-buildpack-nodejs/issues/884

### CODE REVIEW

**Comment by danielleadams** [comment]
@brandonchinn178 I commented on your PR, but I don't think this will fix your issue. Did you test it out?

**Comment by brandonchinn178** [comment]
Yes, my heroku app is configured with `https://github.com/brandonchinn178/heroku-buildpack-nodejs.git` as a buildpack

**Comment by danielleadams** [comment]
Thanks - do you mind adding a changelog entry to this? That's what that last check is failing on.

**Review by edmorley — COMMENTED** [review]
(Switching review to the Node.js SME, to get this out of the languages team review queue)

**Review by joshwlewis — APPROVED** [review]

**Review by joshwlewis — CHANGES_REQUESTED** [review]
I rebased this and triggered CI, but it looks like CI is failing with `ASSERT:Expected STD_ERR to be empty; was </tmp/testHInNN/lib/dependencies.sh: line 150: has_yarn_cache: command not found> expected:<> but was:</tmp/testHInNN/lib/dependencies.sh: line 150: has_yarn_cache: command not found>`. It seems like we still need `has_yarn_cache`. Mind bringing that back?

**Comment by brandonchinn178** [comment]
@joshwlewis done

**Review by joshwlewis — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - Fix typo in conflicting lockfile failure message ([#901](https://github.com/heroku/heroku-buildpack-nodejs/pull/901))
 - Use local packages with Yarn when present ([#913](https://github.com/heroku/heroku-buildpack-nodejs/pull/913))
 - Add support for running "npm ci" when "npm-shrinkwrap.json" is found on the project directory ([#938](https://github.com/heroku/heroku-buildpack-nodejs/pull/938))
+- Support non-zero-install support for Yarn 2 ([#888](https://github.com/heroku/heroku-buildpack-nodejs/pull/888))
```
Comment (edmorley) [inline CHANGELOG.md:10]: @joshwlewis This changelog entry is under the already-released v187 rather than `main`

Comment (joshwlewis) [reply CHANGELOG.md:10]: Correcting this in https://github.com/heroku/heroku-buildpack-nodejs/pull/941/commits/21a6d7e856e16f080ff5990de0f75b722ac36a34.

**Pull request merged**